x/profiles/keeper/msg_server_relationships_test.go

```diff
@@ -3,199 +3,190 @@ package keeper_test
-				suite.testData.user,
-				suite.testData.otherUser,
+				"cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47",
+				"cosmos1cjf97gpzwmaf30pzvaargfgr884mpp5ak8f7ns",
 				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
 			),
 			expErr: false,
 			expEvents: sdk.Events{
 				sdk.NewEvent(
 					types.EventTypeRelationshipCreated,
-					sdk.NewAttribute(types.AttributeRelationshipSender, suite.testData.user),
-					sdk.NewAttribute(types.AttributeRelationshipReceiver, suite.testData.otherUser),
+					sdk.NewAttribute(types.AttributeRelationshipSender, "cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47"),
+					sdk.NewAttribute(types.AttributeRelationshipReceiver, "cosmos1cjf97gpzwmaf30pzvaargfgr884mpp5ak8f7ns"),
 					sdk.NewAttribute(types.AttributeRelationshipSubspace, "4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e"),
 				),
 			},
-			expRelationships: []types.Relationship{
-				types.NewRelationship(
-					suite.testData.user,
-					suite.testData.otherUser,
-					"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
-				),
+			check: func(ctx sdk.Context) {
+				expected := []types.Relationship{
+					types.NewRelationship(
+						"cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47",
+						"cosmos1cjf97gpzwmaf30pzvaargfgr884mpp5ak8f7ns",
+						"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
+					),
+				}
+				suite.Require().Equal(expected, suite.k.GetAllRelationships(ctx))
 			},
 		},
 	}
 
-	for _, test := range tests {
-		suite.SetupTest()
-		suite.Run(test.name, func() {
-
-			profile := suite.CreateProfileFromAddress(suite.testData.user)
-			otherProfile := suite.CreateProfileFromAddress(suite.testData.otherUser)
-
-			err := suite.k.StoreProfile(suite.ctx, profile)
-			suite.Require().NoError(err)
-
-			err = suite.k.StoreProfile(suite.ctx, otherProfile)
-			suite.Require().NoError(err)
-
-			for _, rel := range test.storedRelationships {
-				err := suite.k.SaveRelationship(suite.ctx, rel)
-				suite.Require().NoError(err)
-			}
-
-			for _, block := range test.storedBlock {
-				err := suite.k.SaveUserBlock(suite.ctx, block)
-				suite.Require().NoError(err)
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			ctx, _ := suite.ctx.CacheContext()
+			if tc.store != nil {
+				tc.store(ctx)
 			}
 
 			handler := keeper.NewMsgServerImpl(suite.k)
-			_, err = handler.CreateRelationship(sdk.WrapSDKContext(suite.ctx), test.msg)
+			_, err := handler.CreateRelationship(sdk.WrapSDKContext(ctx), tc.msg)
 
-			if test.expErr {
+			if tc.expErr {
 				suite.Require().Error(err)
 			} else {
 				suite.Require().NoError(err)
-				suite.Require().Equal(test.expEvents, suite.ctx.EventManager().Events())
+				suite.Require().Equal(tc.expEvents, ctx.EventManager().Events())
 
-				stored := suite.k.GetAllRelationships(suite.ctx)
-				suite.Require().Equal(test.expRelationships, stored)
+				if tc.store != nil {
+					tc.store(ctx)
+				}
```

Comment:
```suggestion
				if tc.check!= nil {
					tc.check(ctx)
				}
```
It should be `tc.check(ctx)`.